pants_plugins/sendwave/pants_docker/test_sources.py

```diff
@@ -35,9 +33,9 @@ def sources_runner(rule_runner: RuleRunner) -> RuleRunner:
     """construct rule runner environment with targets containing various
     'Sources' types."""
     build = (
-        "python_library()\n"
```

Comment:
I don't really understand how this was working previously in the absence of a `sources` kwarg. The old docs seem to suggest that [the default for the kwarg was `('`](https://www.pantsbuild.org/v2.2/docs/reference-python_library#codesourcescode) but I don't know what that's supposed to mean 😅 Do you recall how this worked, and if so does adding `source='test.py'` seem like a reasonable step to take as part of the refactor now that [the `source` kwarg is required](https://www.pantsbuild.org/docs/reference-python_source#codesourcecode)?

Reply:
ah python_library used to automatically find all `*.py` files (small caveat: that didn't start with `test`) in the directory - that means you didn't have to specify them explicitly (I think. ... .. that the `python_sources` target in 2.9 does the same thing - (except under the hood it generates a new `python_source` target for each file that it finds instead of having the just a grouping of files as the targets. But I haven't looked into it too much so I might be off.

Having... different classes here does make me a little worried that the new verison of the plugin is not going to be compatible with the old version of pants... which is a little hmmm

Reply:
I think it's true that the plugin won't be compatible with old versions of pants, but I guess anyone who's using it could pin to v0.5 🤷